### PR TITLE
Creating File bundles

### PR DESCRIPTION
**Pull request by Peyman-N**

### CODE REVIEW

File: bids2openminds/main.py [hunk 1]
```diff
@@ -312,11 +312,63 @@ def create_subjects(subject_id, layout_df, layout, collection):
     return subjects_dict, subject_state_dict, subjects_list
 
 
+def create_file_bundle(path, collection, parent_file_bundle=None):
```
Comment (lzehl) [inline bids2openminds/main.py:315]: this seems to run on the full path with no exceptions which means that the dataset folder itself is registered as a file bundle which it should not.

reminder: if you store the openminds results within the dataset directory the code would also register this folder as a filebundle (if run second time) which it should not

File: bids2openminds/main.py [hunk 2]
```diff
@@ -312,11 +312,63 @@ def create_subjects(subject_id, layout_df, layout, collection):
...
+
+    name = str(path).replace("/", "_").replace("\\", "_")
+    if name[0] == "_":
+        name = name[1:]
```
Comment (lzehl) [inline bids2openminds/main.py:319]: reduce to "sub-01_ses-01_anat" (not whole path)

Comment (lzehl) [reply bids2openminds/main.py:319]: this example would mean that there is also a filebundle for "sub-01" and "sub-01_ses-01"

File: bids2openminds/main.py [hunk 3]
```diff
@@ -312,11 +312,63 @@ def create_subjects(subject_id, layout_df, layout, collection):
...
+    openminds_file_bundle = omcore.FileBundle(content_description=f"File bundle created for {str(path)}",
```
Comment (lzehl) [inline bids2openminds/main.py:320]: also reduce to name (with replace _ with /)

File: test/test_file_bundle.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import pytest
+import os
+import openminds.latest.core as omcore
+from openminds import Collection
+from bids2openminds.main import create_file_bundle
```
Comment (lzehl) [inline test/test_file_bundle.py:5]: if test is based on code above the assumed file bundle number is wrong (one less for the not needed dataset file bundle, and another (potentially) one less for the not needed tracking of the openminds folder which does not belong to the BIDS structure)

**Review by lzehl — CHANGES_REQUESTED** [review]

**Comment by apdavison** [comment]
This PR needs more tests:
- test_bids_examples.py should be extended to count file bundles
- test_file_bundle.py should be extended to explicitly test the various scenarios mentioned in #49

**Pull request merged**